### PR TITLE
Fix activation after DEACTIVATE_HMI 

### PR DESCRIPTION
**Pull request by VVeremjova**
According [APPLINK-16172](https://adc.luxoft.com/jira/browse/APPLINK-16172) SDL.ActivateApp should be rejected after DEACTIVATE_HMI
Added error response for HMI

Related to [APPLINK-21868](https://adc.luxoft.com/jira/browse/APPLINK-21868)

### CODE REVIEW

**Comment by VVeremjova** [comment]
Please review @dtrunov, @LuxoftAKutsan

File: src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h [hunk 1]
```diff
@@ -63,9 +63,28 @@ class RequestFromHMI : public CommandImpl, public event_engine::EventObserver {
...
+
+ private:
+  /**
+   * @brief Fills common parameters for SO
+   * @param Contains SO for filling
+   * @param correlation_id the correlation id for the response.
+   * @param function_id the function id for which response will be sent
+   */
+  void FillCommonParametersOfSO(
+      NsSmartDeviceLink::NsSmartObjects::SmartObjectSPtr& message,
```
Comment (dtrunov) [inline src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h]: @VVeremjova Why did you change namespace smart_objects to NsSmartDeviceLink::NsSmartObjects. I think name space "smart_objects" is better.

File: src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h [hunk 2]
```diff
@@ -63,9 +63,28 @@ class RequestFromHMI : public CommandImpl, public event_engine::EventObserver {
...
 
-  void FillCommonParametersOfSO(smart_objects::SmartObject* message,
-                                uint32_t correlation_id,
-                                hmi_apis::FunctionID::eType function_id);
+  /**
+   * @brief SendResponse allows to send error response to hmi
+   * @param correlation_id the correlation id for the response.
+   * @param function_id the function id for which response will be sent
+   * @param result_code the result code.
+   */
```
Comment (dtrunov) [inline src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h]: @VVeremjova missed description for error_message

File: src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h [hunk 3]
```diff
@@ -63,9 +63,28 @@ class RequestFromHMI : public CommandImpl, public event_engine::EventObserver {
...
+  void SendErrorResponse(const uint32_t correlation_id,
+                         const hmi_apis::FunctionID::eType function_id,
+                         const hmi_apis::Common_Result::eType result_code,
+                         std::string error_message);
+
+ private:
+  /**
+   * @brief Fills common parameters for SO
+   * @param Contains SO for filling
```
Comment (dtrunov) [inline src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h]: @VVeremjova Missed param name "message"

Comment (VVeremjova) [inline src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h]: @dtrunov See dde02f4

File: src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h (hunk 2 above)
Comment (VVeremjova) [inline src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h]: @dtrunov See dde02f4

File: src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h (hunk 1 above)
Comment (VVeremjova) [inline src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h]: @dtrunov See dde02f4

File: src/components/application_manager/src/commands/hmi/request_from_hmi.cc [hunk 4]
```diff
@@ -60,13 +60,13 @@ void RequestFromHMI::Run() {}
...
 
-void RequestFromHMI::SendResponse(
-    const bool success,
-    const uint32_t correlation_id,
-    const hmi_apis::FunctionID::eType function_id,
-    const hmi_apis::Common_Result::eType result_code) {
-  smart_objects::SmartObject* message =
-      new smart_objects::SmartObject(smart_objects::SmartType_Map);
+void RequestFromHMI::SendResponse(bool success,
+                                  uint32_t correlation_id,
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/hmi/request_from_hmi.cc]: @VVeremjova why you removed const?

File: src/components/application_manager/src/commands/hmi/request_from_hmi.cc [hunk 5]
```diff
@@ -76,24 +76,26 @@ void RequestFromHMI::SendResponse(
...
+  (*message)[strings::params][hmi_response::code] = result_code;
+
+  (*message)[strings::params][strings::error_msg] = error_message;
+
+  application_manager_.ManageHMICommand(message);
+}
 
 void RequestFromHMI::FillCommonParametersOfSO(
-    smart_objects::SmartObject* message,
+    smart_objects::SmartObjectSPtr& message,
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/hmi/request_from_hmi.cc]: @VVeremjova why you get smart pointer? You can get reference, cant you?

File: src/components/interfaces/HMI_API.xml [hunk 6]
```diff
@@ -2140,21 +2140,6 @@
     <function name="OnUpdateDeviceList" messagetype="notification">
       <description>Notification from HMI to SDL sent when HMI requires update of device list (i.e. when user clicks 'Change Device' button)</description>
     </function>
-    <function name="OnPhoneCall" messagetype="notification">
```
Comment (LuxoftAKutsan) [inline src/components/interfaces/HMI_API.xml]: @VVeremjova you sure that this functions should be removed?

Comment (VVeremjova) [inline src/components/interfaces/HMI_API.xml]: @LuxoftAKutsan Yes, according [APPLINK-18854](https://adc.luxoft.com/jira/browse/APPLINK-18854) these functions were replaced by OnEventChanged (..), but were not removed

File: src/components/application_manager/src/commands/hmi/request_from_hmi.cc (hunk 4 above)
Comment (VVeremjova) [inline src/components/application_manager/src/commands/hmi/request_from_hmi.cc]: @LuxoftAKutsan See 6e3cc19

File: src/components/application_manager/src/commands/hmi/request_from_hmi.cc (hunk 5 above)
Comment (VVeremjova) [inline src/components/application_manager/src/commands/hmi/request_from_hmi.cc]: @LuxoftAKutsan See 6e3cc19

File: src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h [hunk 7]
```diff
@@ -63,9 +63,29 @@ class RequestFromHMI : public CommandImpl, public event_engine::EventObserver {
...
+   * @brief SendResponse allows to send error response to hmi
+   * @param correlation_id the correlation id for the response.
+   * @param function_id the function id for which response will be sent
+   * @param result_code the result code.
+   * @param error_message info message for error.
+   */
+  void SendErrorResponse(const uint32_t correlation_id,
+                         const hmi_apis::FunctionID::eType function_id,
+                         const hmi_apis::Common_Result::eType result_code,
+                         std::string error_message);
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h]: @VVeremjova can it be const ref?

File: src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h [hunk 8]
```diff
@@ -63,9 +63,29 @@ class RequestFromHMI : public CommandImpl, public event_engine::EventObserver {
...
+ private:
+  /**
+   * @brief Fills common parameters for SO
+   * @param message Contains SO for filling
+   * @param correlation_id the correlation id for the response.
+   * @param function_id the function id for which response will be sent
+   */
+  void FillCommonParametersOfSO(
+      NsSmartDeviceLink::NsSmartObjects::SmartObject& message,
+      uint32_t correlation_id,
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h]: @VVeremjova const?

Comment (VVeremjova) [inline src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h]: @LuxoftAKutsan See 5881597

File: src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h (hunk 7 above)
Comment (VVeremjova) [inline src/components/application_manager/include/application_manager/commands/hmi/request_from_hmi.h]: @LuxoftAKutsan See 5881597
